build-tools/packages/build-tools/src/typeValidator/packageJson.ts

```diff
@@ -72,69 +80,172 @@ export async function getPackageDetails(packageDir: string): Promise<PackageDeta
     };
 }
 
+type PreviousVersionStyle =
```

Comment:
Since the exported function below takes this as input, this should probably be exported as well. Also docs? 😁